### PR TITLE
feat: `LAKE_PKG_URL_MAP`

### PR DESCRIPTION
**Pull request by tydeu**
Initial prototype. Overrides are of the form: `LAKE_PKG_URL_MAP={"pkg-name": "git-url", ...}`.

### CODE REVIEW

File: src/lake/Lake/Config/Env.lean [hunk 1]
```diff
@@ -34,20 +37,26 @@ structure Env where
...
     initToolchain := (← IO.getEnv "ELAN_TOOLCHAIN").getD ""
     initLeanPath := ← getSearchPath "LEAN_PATH",
     initLeanSrcPath := ← getSearchPath "LEAN_SRC_PATH",
     initSharedLibPath := ← getSearchPath sharedLibPathEnvVar,
     initPath := ← getSearchPath "PATH"
   }
+where
+  computePkgUrlMap := do
+    let some urlMapStr ← IO.getEnv "LAKE_PKG_URL_MAP" | return {}
+    let .ok urlMap := Json.parse urlMapStr |>.bind fromJson? | return {}
```
Comment (semorrison) [inline src/lake/Lake/Config/Env.lean:58]: This should error, rather than silently ignoring the bad JSON?

File: src/lake/Lake/Load/Main.lean [hunk 2]
```diff
@@ -131,13 +131,13 @@ def buildUpdatedManifest (ws : Workspace)
   match res with
   | (.ok root, deps) =>
     let ws : Workspace ← {ws with root}.finalize
+    let manifest : Manifest := {name? := ws.root.name, packagesDir? := ws.relPkgsDir}
+    let manifest := deps.foldl (·.addPackage ·.manifestEntry) manifest
+    manifest.saveToFile ws.manifestFile
     LakeT.run ⟨ws⟩ <| ws.packages.forM fun pkg => do
       if let some postUpdate := pkg.postUpdate? then
         logInfo s!"{pkg.name}: running post-update hook"
         postUpdate
```
Comment (semorrison) [inline src/lake/Lake/Load/Main.lean:140]: Is the intention here that the post-update hooks might want to rewrite the manifest?

Comment (tydeu) [reply src/lake/Lake/Load/Main.lean:140]: Precisely!

File: src/lake/Lake/Config/Env.lean (hunk 1 above)
Comment (tydeu) [reply src/lake/Lake/Config/Env.lean:58]: This does mean that an improper environment will kill virtually any `lake` command which I was not sure was desirable. But it sounds like that is fine, so will do.

Comment (semorrison) [reply src/lake/Lake/Config/Env.lean:58]: I think given that JSON-via-environment-variable is pretty dodgy to begin with, failing fast is better than the user not noticing a stray comma and not understanding why they can't get their packages.

**Comment by tydeu** [comment]
@semorrison Is the mathlib CI broken on Lean master?  All I did was rebase the PR onto master and it switched from `builds-mathlib` to `breaks-mathlib`.

**Comment by semorrison** [comment]
Sorry, combined Mathlib/Lean CI is really struggling at the moment, due to the number of simultaneous breaking changes we have going.

Since we previously got a green tick, let's decide to ignore subsequent failures.

**Pull request merged**